components/dashboard/src/data/organizations/org-settings-mutation.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { TeamSettings } from "@gitpod/gitpod-protocol";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { getGitpodService } from "../../service/service";
+import { getOrgSettingsQuery, TeamSettingsResult } from "./org-settings-query";
+
+type UpdateTeamSettingsArgs = Pick<TeamSettings, "workspaceSharingDisabled" | "teamId">;
+
+export const useUpdateTeamSettingsMutation = () => {
+    const queryClient = useQueryClient();
+
+    return useMutation<TeamSettings, Error, UpdateTeamSettingsArgs>({
+        mutationFn: async ({ teamId, workspaceSharingDisabled }) => {
+            return await getGitpodService().server.updateTeamSettings(teamId, { workspaceSharingDisabled });
+        },
+        onSuccess: (_, { teamId, workspaceSharingDisabled }) => {
+            const queryKey = getOrgSettingsQuery(teamId);
+            queryClient.setQueryData<TeamSettingsResult>(queryKey, (oldData) => {
+                return {
+                    teamId,
+                    ...oldData,
+                    workspaceSharingDisabled: !workspaceSharingDisabled,
```

Comment:
This looks like we're inverting the value from what was passed into the mutationFn, which is a bit unclear to me why we'd do that. Alternatively, it would probably be best to use the response of the api call if it has this information. If `updateTeamSettings` returns an updated team, we can access that in the first argument to `onSuccess()`, and use that to update the query data with manually.

Reply:
Yeah, this is what I want to ask [internal chat](https://gitpod.slack.com/archives/C01KGM9BH54/p1679933540256649?thread_ts=1679904795.619399&cid=C01KGM9BH54)

I thought first argument of `onSuccess` is the old value because I get this function copied from another one, which said [it is an oldValue](https://github.com/gitpod-io/gitpod/blob/ce3d07e179da0352eb10d8102d377a5b102e2569/components/dashboard/src/data/workspaces/toggle-workspace-pinned-mutation.ts#L26)